src/v/test_utils/scoped_config.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "config/base_property.h"
+#include "config/configuration.h"
+
+// Scoped wrapper around config::shard_local_cfg() that tracks get() calls and,
+// upon destructing, resets any properties that were potentially mutated.
+class scoped_config {
+public:
+    ~scoped_config() {
+        for (auto p : _properties_to_reset) {
+            ss::smp::invoke_on_all([p] {
+                config::shard_local_cfg().get(p).reset();
```

Comment:
i must be confused about how reset works--it resets it back to a default? i would have expected that it reset it back to it's value at the time `get` was called so the scoped configs could compose together or deal with cases where a fixture is customizing a config before a test runs.

Reply:
That's right. I waffled on this a bit. I considered saving the entire config on construction, but felt like that might be a bit heavyweight; or serializing the original value as yaml or json and reverting, but opted for resetting because it was a simpler implementation.

If you're okay with it, I could rename it to some `scoped_config_reset` or somesuch if that makes this less surprising.

Reply:
i think it's probably ok as. no harm in improving/changing it later if it's solving a need now